src/Forms/GridField/GridFieldEditButton.php

```diff
@@ -95,14 +106,63 @@ public function getColumnContent($gridField, $record, $columnName)
         // which can make the form readonly if no edit permissions are available.
 
         $data = new ArrayData(array(
-            'Link' => Controller::join_links($gridField->Link('item'), $record->ID, 'edit')
+            'Link' => Controller::join_links($gridField->Link('item'), $record->ID, 'edit'),
+            'ExtraClass' => $this->getExtraClass()
         ));
 
         $template = SSViewer::get_templates_by_class($this, '', __CLASS__);
         return $data->renderWith($template);
     }
 
     /**
+     * Get the extra HTML classes to add for edit buttons
+     *
+     * @return string
+     */
+    public function getExtraClass()
+    {
+        return implode(' ', Convert::raw2att($this->extraClass));
+    }
+
+    /**
+     * Add an extra HTML class
+     *
+     * @param string $class
+     * @return $this
+     */
+    public function addExtraClass($class = '')
+    {
+        if (!in_array($class, $this->extraClass)) {
+            $this->extraClass[] = $class;
+        }
+        return $this;
+    }
+
+    /**
+     * Checks whether the button has the given HTML class
+     *
+     * @param string $class
+     * @return boolean
+     */
+    public function hasExtraClass($class = '')
```

Comment:
Is this method necessary?

Reply:
For your tests, I'm happy if you just check getExtraClass has the string (or lack thereof)